src/main/java/com/mraof/minestuck/util/PlacementFunctionsUtil.java

```diff
@@ -0,0 +1,233 @@
+	{
+		//TODO placement happens twice(because of two bounding boxes?), one set of stairs can go beyond intended x/z bounds, stairs are fragmented(because it tries to generate in two bounding boxes?)
+		
+		world.setBlockState(minBlockPosIn, Blocks.GOLD_BLOCK.getDefaultState(), Constants.BlockFlags.BLOCK_UPDATE);
+		world.setBlockState(maxBlockPosIn, Blocks.DIAMOND_BLOCK.getDefaultState(), Constants.BlockFlags.BLOCK_UPDATE);
+		
+		//Debug.debugf("createPlainSpiralStaircase. boundingBox = %s", boundingBox1);
+		int xIterator = minBlockPosIn.getX();
+		
+		int zIterator = minBlockPosIn.getZ();
+		
+		boolean isXIterating = true; //toggles between moving in x or z direction
+		boolean isXIterateReversed = false; //moving in negative x direction if positive
+		boolean isZIterateReversed = false; //moving in negative z direction if positive
+		for(int yIterator = minBlockPosIn.getY(); yIterator <= maxBlockPosIn.getY(); yIterator++)
+		{
+			BlockPos iteratorPos = new BlockPos(xIterator, yIterator, zIterator);
+			
+			/*Debug.debugf("createPlainSpiralStaircase. isXIterating = %s, isPosInsideBounding = %s, iteratorPos = %s",
+					isXIterating, boundingBox1.isVecInside(iteratorPos), iteratorPos);*/
+			
+			if(xIterator >= maxBlockPosIn.getX() && !isXIterateReversed) //moving positive x
+			{
+				isXIterating = false;
+				isXIterateReversed = true;
+			} else if(zIterator >= maxBlockPosIn.getZ() && !isZIterateReversed) //moving positive z
+			{
+				isXIterating = true;
+				isZIterateReversed = true;
+			} else if(xIterator <= minBlockPosIn.getX() && isXIterateReversed) //moving negative x
+			{
+				isXIterating = false;
+				isXIterateReversed = false;
+			} else if(zIterator <= minBlockPosIn.getZ() && isZIterateReversed) //moving negative z
+			{
+				isXIterating = true;
+				isZIterateReversed = true;
+			}
+			
+			if(boundingBox1.isVecInside(iteratorPos) || boundingBox2.isVecInside(iteratorPos))
+			{
+				//Debug.debugf("createPlainSpiralStaircase. placed at %s", iteratorPos);
+				world.setBlockState(iteratorPos, blockState, Constants.BlockFlags.BLOCK_UPDATE);
+				
+				if(isXIterating && !isXIterateReversed)
+				{
+					xIterator++;
+				} else if(!isXIterating && !isZIterateReversed)
+				{
+					zIterator++;
+				} else if(isXIterating && isXIterateReversed)
+				{
+					xIterator--;
+				} else if(!isXIterating && isZIterateReversed)
+				{
+					zIterator--;
+				}
+				/*
+				if(isXIterating && !isXIterateReversed)
+				{
+					xIterator++;
+				}
+				if(!isXIterating && !isZIterateReversed)
+				{
+					zIterator++;
+				}
+				if(isXIterating && isXIterateReversed)
+				{
+					xIterator--;
+				}
+				if(!isXIterating && isZIterateReversed)
+				{
+					zIterator--;
+				}
+				 */
+			}
+			
+			if(yIterator >= maxBlockPosIn.getY())
+			{
+				break;
+			}
+		}
+	}
```

Comment:
Im not sure how to fix the issue with the staircase generating in a fragmented manner. It does seem related to bounding boxes but if thats the case I will probably need help